### ISSUE TITLE
Cost(loss) result using placeholder vs. primitive list

### ISSUE THREAD

**Issue by syenpark** [hunkim/DeepLearningZeroToAll#178]
I spotted cost result of 'lab-02-2-linear_regression_feed.py' is slightly different in using placeholder and using a python list.

```
x_train = [1, 2, 3]
y_train = [1, 2, 3]

# Our hypothesis XW+b
hypothesis = x_train * W + b

# cost/loss function
cost = tf.reduce_mean(tf.square(hypothesis - y_train))
```

> 0 **_2.82329_** [ 2.12867713] [-0.85235667]
> 20 _**0.190351**_ [ 1.53392804] [-1.05059612]
> ...
> 1980 **_1.32962e-05_** [ 1.00423515] [-0.00962736]
> 2000 **_1.20761e-05_** [ 1.00403607] [-0.00917497]
>
> [ 5.0110054]
> [ 2.50091505]
> [ 1.49687922  3.50495124]

```
X = tf.placeholder(tf.float32, shape=[None])
Y = tf.placeholder(tf.float32, shape=[None])

# Our hypothesis XW+b
hypothesis = X * W + b

# cost/loss function
cost = tf.reduce_mean(tf.square(hypothesis - Y))

sess.run([cost, W, b, train], feed_dict={X: [1, 2, 3], Y: [1, 2, 3]})
```
> 0 _**3.52408**_ [ 2.12867713] [-0.85235667]
> 20 **_0.197499_** [ 1.53392804] [-1.05059612]
> 1960 **_1.47111e-05_** [ 1.004444] [-0.01010205]
> 1980 **_1.3361e-05_** [ 1.00423515] [-0.00962736]
> ...
> 2000 **_1.21343e-05_** [ 1.00403607] [-0.00917497]
>
> [ 5.0110054]
> [ 2.50091505]
> [ 1.49687922  3.50495124]
 

**_Hypothesis_**, **_W_**, and **_b_** are the same, while **_cost_** result is different even with those same values. I suppose this cost result should be equal regardless of node type, so that I cannot understand this discrepancy in cost and conformability in others in spite of that.  Did I something miss?

**Comment by kkweon** [comment]
W and b are learnable parameters and these values at optimal or at the final step are not unique, so it is normal that values are slightly different.

**Comment by syenpark** [comment]
@kkweon, thanks for your kindly help! I will double check whether I missed or not. Should others also have the same problem, please let me know!
